src/base.py

```diff
@@ -35,20 +40,22 @@ class HTMLParser(ABC):
 
```

Comment:
similar comment to the above - a shared interface for `Parsers` might be nice...